### ISSUE TITLE
No jump to definition in assembly (& an outdated dep)

### ISSUE THREAD

**Issue by lincolnauster** [jmacdonald/amp#249]
Expected behavior: In assembly files  (.s and .S), amp should be able to jump to the definition of labels, which look like `some_label:`. \
Actual behavior: There are no definitions to jump to (though syntax highlighting is fine).

It looks like jump to definition is 99% syntect's problem, but as it turns out amp and scribe are about 2 major versions behind current stable syntect (2.1.0 -> 4.5.0), and should probably be updated *before* messing with its assembly lexing. It looks like christoph-heiss has a PR for this in jmacdonald/scribe#18, which would then have to be propogated over to amp (wich is something I might be able to help out with over this winter break). Is that PR in a state to merge? Looking over the changes, it doesn't seem too onerous given that it sort of *has* to break the API.

tl;dr: this is a nice-to-have that should be looked at after updating syntect.
